backend/modules/student/routers/student_instances.py

```
@@ -291,3 +398,366 @@ async def get_student_simulation_instance(
+                StudentSimulationInstance.student_id == current_user.id
+            ).first()
+        
+        if not instance:
+            # Try to find if instance exists at all (for debugging) - use a fresh query
+            db.expire_all()
+            instance_by_id = db.query(StudentSimulationInstance).filter(
+                StudentSimulationInstance.unique_id == instance_unique_id
+            ).first()
+            if instance_by_id:
+                logger.warning(f"Instance {instance_unique_id} exists but belongs to student_id={instance_by_id.student_id}, not {current_user.id}")
+                raise HTTPException(status_code=403, detail="This simulation instance belongs to a different student")
```

Ruff (0.14.8)

Abstract `raise` to an inner function

(TRY301)